crates/bevy_ecs/src/system/query.rs

```diff
@@ -576,15 +605,11 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> Query<'w, 's, Q, F> {
         )
     }
 
-    /// Runs `f` on each query result. This is faster than the equivalent iter() method, but cannot
-    /// be chained like a normal [`Iterator`].
-    ///
-    /// This can only pass in immutable data, see [`Self::for_each_mut`] for mutable access.
+    /// Runs `f` on each read-only query item.
```

Comment:
Is there value in keeping the note on "faster than iter()"?

Reply:
I think not, #5741 has a dedicated performance section.